code/email/MailblockMailSubscriber.php

```diff
@@ -2,65 +2,70 @@
 
 namespace Mailblock\Email;
 
-use SilverStripe\SiteConfig\SiteConfig;
-use SilverStripe\Control\Email\Email;
-use SilverStripe\Control\Email\SwiftPlugin;
 use SilverStripe\Control\Director;
+use SilverStripe\Control\Email\Email;
+use SilverStripe\SiteConfig\SiteConfig;
 use SilverStripe\Subsites\Model\Subsite;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Mailer\Event\MessageEvent;
 
 /**
- * Mail plugin to rewrite email recipients depending on Mailblock SiteConfig settings.
+ * See https://symfony.com/doc/current/mailer.html#mailer-events for further info
  */
-class MailblockPlugin extends SwiftPlugin
+class MailSubscriber implements EventSubscriberInterface
 {
-    /**
-     * Before sending a message make sure all our overrides are taken into account
-     *
-     * @param \Swift_Events_SendEvent $evt
-     */
-    public function beforeSendPerformed(\Swift_Events_SendEvent $evt)
+    public static function getSubscribedEvents()
+    {
+        return [
+            MessageEvent::class => 'onMessage',
+        ];
+    }
+
+    public function onMessage(MessageEvent $event): void
     {
-        /** @var \Swift_Message $message */
-        $message = $evt->getMessage();
+        /** @var Email $message */
+        $message = $event->getMessage();
```

Comment:
Technically it's a `Symfony\Component\Mime\RawMessage` which `Email` is now a subclass of - but the main mail event subscriber will already throw an exception if it's not specifically an `Email` instance anyway so we don't have to check that here.